Create 'install_tensorflow'

We need to create `install_tensorflow`, while simultaneously verifying that our [installed](https://github.com/jeff1evesque/ist-664/pull/5#issuecomment-430469327) cuda + cudnn is accessible with tensorflow.

This is my build `make.log`:

```bash
[root@xxx-xxx-xxx-xxx centos]# cat /var/lib/dkms/nvidia/384.98/build/make.log
DKMS make.log for nvidia-384.98 for kernel 3.10.0-862.14.4.el7.x86_64 (x86_64)
Thu Oct 18 01:00:19 UTC 2018
make "CC=cc" KBUILD_OUTPUT=/lib/modules/3.10.0-862.14.4.el7.x86_64/build KBUILD_VERBOSE= -C /lib/modules/3.10.0-862.14.4.el7.x86_64/source M=/var/lib/dkms/nvidia/384.98/build ARCH=x86_64 NV_KERNEL_SOURCES=/lib/modules/3.10.0-862.14.4.el7.x86_64/source NV_KERNEL_OUTPUT=/lib/modules/3.10.0-862.14.4.el7.x86_64/build NV_KERNEL_MODULES="nvidia nvidia-uvm nvidia-modeset nvidia-drm" INSTALL_MOD_DIR=kernel/drivers/video modules
make[1]: Entering directory `/usr/src/kernels/3.10.0-862.14.4.el7.x86_64'
make -C /lib/modules/3.10.0-862.14.4.el7.x86_64/build \
KBUILD_SRC=/usr/src/kernels/3.10.0-862.14.4.el7.x86_64 \
KBUILD_EXTMOD="/var/lib/dkms/nvidia/384.98/build" -f /usr/src/kernels/3.10.0-862.14.4.el7.x86_64/Makefile \
modules
  SYMLINK /var/lib/dkms/nvidia/384.98/build/nvidia/nv-kernel.o
  SYMLINK /var/lib/dkms/nvidia/384.98/build/nvidia-modeset/nv-modeset-kernel.o
 CONFTEST: remap_pfn_range
 CONFTEST: INIT_WORK
 CONFTEST: hash__remap_4k_pfn
 CONFTEST: follow_pfn
 CONFTEST: vmap
 CONFTEST: set_pages_uc
 CONFTEST: set_memory_uc
 CONFTEST: set_memory_array_uc
 CONFTEST: change_page_attr
 CONFTEST: pci_get_class
 CONFTEST: pci_choose_state
 CONFTEST: vm_insert_page
 CONFTEST: acpi_device_id
 CONFTEST: acquire_console_sem
 CONFTEST: console_lock
 CONFTEST: kmem_cache_create
 CONFTEST: on_each_cpu
 CONFTEST: smp_call_function
 CONFTEST: acpi_evaluate_integer
 CONFTEST: ioremap_cache
 CONFTEST: ioremap_wc
 CONFTEST: acpi_walk_namespace
 CONFTEST: pci_domain_nr
 CONFTEST: pci_dma_mapping_error
 CONFTEST: sg_alloc_table
 CONFTEST: sg_init_table
 CONFTEST: pci_get_domain_bus_and_slot
 CONFTEST: get_num_physpages
 CONFTEST: efi_enabled
 CONFTEST: proc_create_data
 CONFTEST: pde_data
 CONFTEST: proc_remove
 CONFTEST: pm_vt_switch_required
 CONFTEST: xen_ioemu_inject_msi
 CONFTEST: phys_to_dma
 CONFTEST: get_dma_ops
 CONFTEST: write_cr4
 CONFTEST: of_get_property
 CONFTEST: of_find_node_by_phandle
 CONFTEST: of_node_to_nid
 CONFTEST: pnv_pci_get_npu_dev
 CONFTEST: for_each_online_node
 CONFTEST: node_end_pfn
 CONFTEST: pci_bus_address
 CONFTEST: pci_stop_and_remove_bus_device
 CONFTEST: pci_remove_bus_device
 CONFTEST: request_threaded_irq
 CONFTEST: register_cpu_notifier
 CONFTEST: cpuhp_setup_state
 CONFTEST: remap_page_range
 CONFTEST: address_space_init_once
 CONFTEST: kbasename
 CONFTEST: fatal_signal_pending
 CONFTEST: list_cut_position
 CONFTEST: vzalloc
 CONFTEST: wait_on_bit_lock_argument_count
 CONFTEST: bitmap_clear
 CONFTEST: usleep_range
 CONFTEST: radix_tree_empty
 CONFTEST: drm_dev_unref
 CONFTEST: drm_reinit_primary_mode_group
 CONFTEST: drm_atomic_set_mode_for_crtc
 CONFTEST: drm_atomic_clean_old_fb
 CONFTEST: get_user_pages_remote
 CONFTEST: drm_gem_object_lookup
 CONFTEST: drm_atomic_state_free
 CONFTEST: drm_driver_has_gem_prime_res_obj
 CONFTEST: is_export_symbol_gpl_of_node_to_nid
 CONFTEST: i2c_adapter
 CONFTEST: pm_message_t
 CONFTEST: irq_handler_t
 CONFTEST: acpi_device_ops
 CONFTEST: acpi_op_remove
 CONFTEST: outer_flush_all
 CONFTEST: proc_dir_entry
 CONFTEST: scatterlist
 CONFTEST: sg_table
 CONFTEST: file_operations
 CONFTEST: vm_operations_struct
 CONFTEST: atomic_long_type
 CONFTEST: pci_save_state
 CONFTEST: file_inode
 CONFTEST: task_struct
 CONFTEST: kuid_t
 CONFTEST: dma_ops
 CONFTEST: dma_map_ops
 CONFTEST: noncoherent_swiotlb_dma_ops
 CONFTEST: vm_fault_present
 CONFTEST: vm_fault_has_address
 CONFTEST: kernel_write
 CONFTEST: strnstr
 CONFTEST: iterate_dir
 CONFTEST: kstrtoull
 CONFTEST: fault_flags
 CONFTEST: atomic64_type
 CONFTEST: address_space
 CONFTEST: backing_dev_info
 CONFTEST: mm_context_t
 CONFTEST: vm_ops_fault_removed_vma_arg
 CONFTEST: pnv_npu2_init_context
 CONFTEST: drm_bus_present
 CONFTEST: drm_bus_has_bus_type
 CONFTEST: drm_bus_has_get_irq
 CONFTEST: drm_bus_has_get_name
 CONFTEST: drm_driver_has_legacy_dev_list
 CONFTEST: drm_driver_has_set_busid
 CONFTEST: drm_crtc_state_has_connectors_changed
 CONFTEST: drm_init_function_args
 CONFTEST: drm_mode_connector_list_update_has_merge_type_bits_arg
 CONFTEST: drm_helper_mode_fill_fb_struct
 CONFTEST: drm_master_drop_has_from_release_arg
 CONFTEST: drm_mode_config_funcs_has_atomic_state_alloc
 CONFTEST: drm_driver_unload_has_int_return_type
 CONFTEST: drm_crtc_helper_funcs_has_atomic_enable
 CONFTEST: dom0_kernel_present
 CONFTEST: nvidia_vgpu_kvm_build
 CONFTEST: nvidia_grid_build
 CONFTEST: drm_available
 CONFTEST: drm_atomic_available
 CONFTEST: drm_atomic_modeset_nonblocking_commit_available
 CONFTEST: is_export_symbol_gpl_refcount_inc
 CONFTEST: is_export_symbol_gpl_refcount_dec_and_test
  CC [M]  /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.o
  CC [M]  /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.o
  CC [M]  /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.o
  CC [M]  /var/lib/dkms/nvidia/384.98/build/nvidia/nv.o
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h: In function ‘NV_GET_USER_PAGES’:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: warning: passing argument 1 of ‘get_user_pages’ makes pointer from integer without a cast [enabled by default]
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: expected ‘struct task_struct *’ but argument is of type ‘long unsigned int’
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: warning: passing argument 2 of ‘get_user_pages’ makes pointer from integer without a cast [enabled by default]
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: expected ‘struct mm_struct *’ but argument is of type ‘long unsigned int’
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: warning: passing argument 4 of ‘get_user_pages’ makes integer from pointer without a cast [enabled by default]
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: expected ‘long unsigned int’ but argument is of type ‘struct page **’
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: warning: passing argument 5 of ‘get_user_pages’ makes integer from pointer without a cast [enabled by default]
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: expected ‘int’ but argument is of type ‘struct vm_area_struct **’
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: error: too few arguments to function ‘get_user_pages’
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: declared here
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h: In function ‘NV_GET_USER_PAGES_REMOTE’:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:97:45: warning: passing argument 6 of ‘get_user_pages_remote’ makes integer from pointer without a cast [enabled by default]
                                             pages, vmas);
                                             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1377:6: note: expected ‘int’ but argument is of type ‘struct page **’
 long get_user_pages_remote(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:97:45: warning: passing argument 7 of ‘get_user_pages_remote’ from incompatible pointer type [enabled by default]
                                             pages, vmas);
                                             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1377:6: note: expected ‘struct page **’ but argument is of type ‘struct vm_area_struct **’
 long get_user_pages_remote(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:97:45: error: too few arguments to function ‘get_user_pages_remote’
                                             pages, vmas);
                                             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1377:6: note: declared here
 long get_user_pages_remote(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h: In function ‘NV_GET_USER_PAGES’:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: warning: passing argument 1 of ‘get_user_pages’ makes pointer from integer without a cast [enabled by default]
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: expected ‘struct task_struct *’ but argument is of type ‘long unsigned int’
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: warning: passing argument 2 of ‘get_user_pages’ makes pointer from integer without a cast [enabled by default]
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: expected ‘struct mm_struct *’ but argument is of type ‘long unsigned int’
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: warning: passing argument 4 of ‘get_user_pages’ makes integer from pointer without a cast [enabled by default]
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: expected ‘long unsigned int’ but argument is of type ‘struct page **’
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: warning: passing argument 5 of ‘get_user_pages’ makes integer from pointer without a cast [enabled by default]
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: expected ‘int’ but argument is of type ‘struct vm_area_struct **’
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: error: too few arguments to function ‘get_user_pages’
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: declared here
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h: In function ‘NV_GET_USER_PAGES_REMOTE’:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:97:45: warning: passing argument 6 of ‘get_user_pages_remote’ makes integer from pointer without a cast [enabled by default]
                                             pages, vmas);
                                             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1377:6: note: expected ‘int’ but argument is of type ‘struct page **’
 long get_user_pages_remote(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:97:45: warning: passing argument 7 of ‘get_user_pages_remote’ from incompatible pointer type [enabled by default]
                                             pages, vmas);
                                             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1377:6: note: expected ‘struct page **’ but argument is of type ‘struct vm_area_struct **’
 long get_user_pages_remote(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:97:45: error: too few arguments to function ‘get_user_pages_remote’
                                             pages, vmas);
                                             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1377:6: note: declared here
 long get_user_pages_remote(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h: In function ‘NV_GET_USER_PAGES’:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: warning: passing argument 1 of ‘get_user_pages’ makes pointer from integer without a cast [enabled by default]
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: expected ‘struct task_struct *’ but argument is of type ‘long unsigned int’
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: warning: passing argument 2 of ‘get_user_pages’ makes pointer from integer without a cast [enabled by default]
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: expected ‘struct mm_struct *’ but argument is of type ‘long unsigned int’
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: warning: passing argument 4 of ‘get_user_pages’ makes integer from pointer without a cast [enabled by default]
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: expected ‘long unsigned int’ but argument is of type ‘struct page **’
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: warning: passing argument 5 of ‘get_user_pages’ makes integer from pointer without a cast [enabled by default]
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: expected ‘int’ but argument is of type ‘struct vm_area_struct **’
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: error: too few arguments to function ‘get_user_pages’
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: declared here
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h: In function ‘NV_GET_USER_PAGES_REMOTE’:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:97:45: warning: passing argument 6 of ‘get_user_pages_remote’ makes integer from pointer without a cast [enabled by default]
                                             pages, vmas);
                                             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1377:6: note: expected ‘int’ but argument is of type ‘struct page **’
 long get_user_pages_remote(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:97:45: warning: passing argument 7 of ‘get_user_pages_remote’ from incompatible pointer type [enabled by default]
                                             pages, vmas);
                                             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1377:6: note: expected ‘struct page **’ but argument is of type ‘struct vm_area_struct **’
 long get_user_pages_remote(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv.c:13:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:97:45: error: too few arguments to function ‘get_user_pages_remote’
                                             pages, vmas);
                                             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv.c:13:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1377:6: note: declared here
 long get_user_pages_remote(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.c:24:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h: In function ‘NV_GET_USER_PAGES’:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: warning: passing argument 1 of ‘get_user_pages’ makes pointer from integer without a cast [enabled by default]
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.c:24:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: expected ‘struct task_struct *’ but argument is of type ‘long unsigned int’
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.c:24:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: warning: passing argument 2 of ‘get_user_pages’ makes pointer from integer without a cast [enabled by default]
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.c:24:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: expected ‘struct mm_struct *’ but argument is of type ‘long unsigned int’
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.c:24:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: warning: passing argument 4 of ‘get_user_pages’ makes integer from pointer without a cast [enabled by default]
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.c:24:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: expected ‘long unsigned int’ but argument is of type ‘struct page **’
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.c:24:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: warning: passing argument 5 of ‘get_user_pages’ makes integer from pointer without a cast [enabled by default]
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.c:24:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: expected ‘int’ but argument is of type ‘struct vm_area_struct **’
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.c:24:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:70:13: error: too few arguments to function ‘get_user_pages’
             return get_user_pages(start, nr_pages, flags, pages, vmas);
             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.c:24:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1381:6: note: declared here
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.c:24:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h: In function ‘NV_GET_USER_PAGES_REMOTE’:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:97:45: warning: passing argument 6 of ‘get_user_pages_remote’ makes integer from pointer without a cast [enabled by default]
                                             pages, vmas);
                                             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.c:24:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1377:6: note: expected ‘int’ but argument is of type ‘struct page **’
 long get_user_pages_remote(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.c:24:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:97:45: warning: passing argument 7 of ‘get_user_pages_remote’ from incompatible pointer type [enabled by default]
                                             pages, vmas);
                                             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.c:24:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1377:6: note: expected ‘struct page **’ but argument is of type ‘struct vm_area_struct **’
 long get_user_pages_remote(struct task_struct *tsk, struct mm_struct *mm,
      ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:21:0,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.c:24:
/var/lib/dkms/nvidia/384.98/build/common/inc/nv-mm.h:97:45: error: too few arguments to function ‘get_user_pages_remote’
                                             pages, vmas);
                                             ^
In file included from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-pgprot.h:17:0,
                 from /var/lib/dkms/nvidia/384.98/build/common/inc/nv-linux.h:20,
                 from /var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.c:24:
/usr/src/kernels/3.10.0-862.14.4.el7.x86_64/include/linux/mm.h:1377:6: note: declared here
 long get_user_pages_remote(struct task_struct *tsk, struct mm_struct *mm,
      ^
make[3]: *** [/var/lib/dkms/nvidia/384.98/build/nvidia/nv-frontend.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: *** [/var/lib/dkms/nvidia/384.98/build/nvidia/nv-instance.o] Error 1
make[3]: *** [/var/lib/dkms/nvidia/384.98/build/nvidia/nv-gpu-numa.o] Error 1
make[3]: *** [/var/lib/dkms/nvidia/384.98/build/nvidia/nv.o] Error 1
make[2]: *** [_module_/var/lib/dkms/nvidia/384.98/build] Error 2
make[1]: *** [sub-make] Error 2
make[1]: Leaving directory `/usr/src/kernels/3.10.0-862.14.4.el7.x86_64'
make: *** [modules] Error 2
```

--- a/utility/install_cuda
+++ b/utility/install_cuda
@@ -5,12 +5,15 @@
 ##
 ##     - https://www.tensorflow.org/install/source#tested_source_configurations
 ##
-CUDA_VERSION=9
-NVIDIA_CUDA_URL='https://developer.nvidia.com/compute/cuda'
-REPO="rhel7-$CUDA_VERSION-0-local-$CUDA_VERSION.0.176-1.x86_64-rpm"
+BASE_URL='http://developer.download.nvidia.com'
+
+CUDA_VERSION=9.2
+CUDA_VERSIOND="${CUDA_VERSION:0:1}-${CUDA_VERSION:2:3}"
+CUDA_URL='https://developer.nvidia.com/compute/cuda'
+CUDA_REPO="cuda-repo-rhel7-$CUDA_VERSIOND-local-$CUDA_VERSION.148-1.x86_64"
 
 CUDNN_VERSION=7.2.1
-CUDNN_URL='http://developer.download.nvidia.com/compute/redist/cudnn'
+CUDNN_URL="$BASE_URL/compute/redist/cudnn"
 CUDNN_FILE="cudnn-9.2-linux-x64-v$CUDNN_VERSION.38.tgz"
 
 ##
@@ -20,27 +23,21 @@ CUDNN_FILE="cudnn-9.2-linux-x64-v$CUDNN_VERSION.38.tgz"
 ##     'wget -c <filename>' can be reimplemented to resume the download.
 ##
 sudo yum -y install wget
-wget "$NVIDIA_CUDA_URL/$CUDA_VERSION.0/Prod/local_installers/cuda-repo-$REPO"\
-    -O "$REPO.rpm"
+wget "$CUDA_URL/$CUDA_VERSION/Prod2/local_installers/$CUDA_REPO"\
+    -O "$CUDA_REPO.rpm"
 
 ## clear cache
 sudo yum clean all
 sudo rm -rf /var/cache/yum
 
-## install nvidia drivers
+## nvidia prerequisites
 sudo yum -y update
 sudo yum -y groupinstall 'Development Tools'
-sudo yum -y install kernel-devel epel-release dkms
+sudo yum -y install kernel-devel kernel-headers
 
 ## install package
-sudo rpm -i "$REPO.rpm"
+sudo rpm -i "$CUDA_REPO.rpm"
 sudo yum install -y cuda
-
-## append to path
-PATHOK=$(grep cuda /etc/environment)
-if [ -z $PATHOK ]; then
-    sed -e "/^PATH/s/\"$/:\/usr\/local\/cuda-$CUDA_VERSION\/bin\"/g" /etc/environment
-fi
 
 ## download cuDNN
 wget "$CUDNN_URL/v$CUDNN_VERSION/$CUDNN_FILE"

--- a/utility/install_tensorflow
+++ b/utility/install_tensorflow
@@ -1,0 +1,35 @@
+#!/bin/bash
+
+## local variables
+PYTHON_VERSION=36
+PYTHON_VERSIOND=${PYTHON_VERSION:0:1}.${PYTHON_VERSION:1:2}
+TENSORFLOW_VERSION=1.10.0
+
+## add repository
+sudo yum install -y https://centos7.iuscommunity.org/ius-release.rpm
+sudo yum update
+
+## install python3
+sudo yum install -y "python${PYTHON_VERSION}u"\
+    "python${PYTHON_VERSION}u-libs"\
+    "python${PYTHON_VERSION}u-devel"\
+    "python${PYTHON_VERSION}u-pip"
+
+echo '====================================================='
+echo "python installed: $(python$PYTHON_VERSIOND -V)"
+echo '====================================================='
+
+## temporary pip3: symlink did not work
+PIP3_PATH=$(which pip$PYTHON_VERSIOND)
+cp -f "$PIP3_PATH" /usr/bin/pip3
+
+## install tensorflow
+sudo pip3 install tensorflow-gpu=="$TENSORFLOW_VERSION"
+
+echo '====================================================='
+echo 'tensorflow installed:'
+echo "$(pip$PYTHON_VERSIOND show tensorflow-gpu)"
+echo '====================================================='
+
+## remove copied pip3
+rm -f /usr/bin/pip3

